python/cudf/cudf/core/column/column.py

```diff
@@ -701,11 +703,26 @@ def to_pandas(
             raise ValueError(
                 f"{arrow_type=} and {nullable=} cannot both be set."
             )
-        elif nullable:
-            raise NotImplementedError(f"{nullable=} is not implemented.")
         pa_array = self.to_arrow()
-        if arrow_type:
+
+        # Check if dtype is an ArrowDtype or pd.ExtensionDtype subclass
+        if arrow_type or isinstance(self.dtype, pd.ArrowDtype):
             return pd.Index(pd.arrays.ArrowExtensionArray(pa_array))
+        elif (
+            (
+                nullable
+                or isinstance(self.dtype, pd.api.extensions.ExtensionDtype)
+            )
+            and isinstance(
+                pandas_nullable_dtype := np_dtypes_to_pandas_dtypes.get(
+                    self.dtype, self.dtype
+                ),
+                pd.api.extensions.ExtensionDtype,
+            )
+            and hasattr(pandas_nullable_dtype, "__from_arrow__")
```

Comment:
```suggestion
```
If we check the exact subclasses as suggested above, I don't think we need this check since they all implement `__from_arrow__`

Reply:
Yup, dropped this.